src/sietch/distributed_dune.ml

```diff
@@ -25,6 +25,11 @@ let make cache config = { cache; clients = Clients.empty; config }
 
 let debug = Dune_util.Log.info
 
+let hash h =
```

Comment:
As mentioned in the other review, I'd prefer the name to reflect what's happening. Something like `digest_of_hex_exn` would make it self-documenting (otherwise, reading `hash h` is pretty difficult).